### PR TITLE
SPRINT 2 RELEASE

### PR DESCRIPTION
**Pull request by RafterGit**
## Overview
PR to merge the user stories that were completed in Sprint 2. 

### List of completed user stories
* [As a user I need to log in (frontend)](https://app.zenhub.com/workspaces/tdrs-sprint-board-5f18ab06dfd91c000f7e682e/issues/raft-tech/tanf-app/56)
* [As a user, I can log out from the UI (mostly frontend)](https://app.zenhub.com/workspaces/tdrs-sprint-board-5f18ab06dfd91c000f7e682e/issues/raft-tech/tanf-app/61)
* [As a user I need to log in (backend)](https://app.zenhub.com/workspaces/tdrs-sprint-board-5f18ab06dfd91c000f7e682e/issues/raft-tech/tanf-app/57)
* [As a developer I need scaffolding for the backend so I can contribute to the project](https://app.zenhub.com/workspaces/tdrs-sprint-board-5f18ab06dfd91c000f7e682e/issues/raft-tech/tanf-app/2)
* [UI Scaffolding Updates based on PR feedback](https://app.zenhub.com/workspaces/tdrs-sprint-board-5f18ab06dfd91c000f7e682e/issues/raft-tech/tanf-app/112)
* [Set up CI/CD pipeline for OWASP Zap, Pa11y, and Code Coverage Reporting](https://app.zenhub.com/workspaces/tdrs-sprint-board-5f18ab06dfd91c000f7e682e/issues/raft-tech/tanf-app/18)
* [Design mockup for a new user to be assigned an existing role](https://app.zenhub.com/workspaces/tdrs-sprint-board-5f18ab06dfd91c000f7e682e/issues/raft-tech/tanf-app/164):lock:
* [Design mockup for a new user to be assigned an existing role](https://app.zenhub.com/workspaces/tdrs-sprint-board-5f18ab06dfd91c000f7e682e/issues/raft-tech/tanf-app/164) :lock:
* [Create mockups for responsive version of the app to view on my phone/tablet](https://app.zenhub.com/workspaces/tdrs-sprint-board-5f18ab06dfd91c000f7e682e/issues/raft-tech/tanf-app/178):lock:
* [Conduct OFA Interviews](https://app.zenhub.com/workspaces/tdrs-sprint-board-5f18ab06dfd91c000f7e682e/issues/raft-tech/tanf-app/51):lock:
* [Discuss and finalize sprint report details](https://app.zenhub.com/workspaces/tdrs-sprint-board-5f18ab06dfd91c000f7e682e/issues/raft-tech/tanf-app/185)
* [Document review process for future issues](https://app.zenhub.com/workspaces/tdrs-sprint-board-5f18ab06dfd91c000f7e682e/issues/raft-tech/tanf-app/41)
* [Follow-up from Sprint Retro 8/18](https://app.zenhub.com/workspaces/tdrs-sprint-board-5f18ab06dfd91c000f7e682e/issues/raft-tech/tanf-app/181)

### List of user stories moving to next sprint
* [Compile a list of states to learn from and interview for Round 2](https://app.zenhub.com/workspaces/tdrs-sprint-board-5f18ab06dfd91c000f7e682e/issues/raft-tech/tanf-app/163)
* [Synthesize feedback from research sessions with OFA (August 2020)](https://app.zenhub.com/workspaces/tdrs-sprint-board-5f18ab06dfd91c000f7e682e/issues/raft-tech/tanf-app/113)
* [Create a user research plan for STTs for Sept 2020](https://app.zenhub.com/workspaces/tdrs-sprint-board-5f18ab06dfd91c000f7e682e/issues/raft-tech/tanf-app/63)
* [As a user who is not yet logged in, I see a landing page based on the design](https://app.zenhub.com/workspaces/tdrs-sprint-board-5f18ab06dfd91c000f7e682e/issues/raft-tech/tanf-app/172)
* [Get a tour of OFA resource mailbox to determine whether access is needed](https://app.zenhub.com/workspaces/tdrs-sprint-board-5f18ab06dfd91c000f7e682e/issues/raft-tech/tanf-app/167)

### User research artifacts delivered:
Mockups:
* [Assign a new user an existing role](https://jstrieb.github.io/link-lock/#eyJ2IjoiMC4wLjEiLCJlIjoiS1AwM2RYMU5mbXRHNXRKYWNsakdZaXVLRk9iQTl3cXNNL1RDaC85UWZJWURzUmFLSkUwTEJqbUgwODRYdGdCMnE4WU5PUHdOQVNGKzRESkQ1SlhwTzZtZWNQcTlQWnNhcURJS1ZVZHdPc2hDL1ZOb29tWjlJOHQ3SXBWS25QTm1zV0lINHhWNU5aR0pIYklsNnVtSms5eVFhbzg9IiwiaSI6IkZIbDREcmFzQVJKT29IbUsifQ==) :lock: 
* [New user creates a profile](https://jstrieb.github.io/link-lock/#eyJ2IjoiMC4wLjEiLCJlIjoiUit1TEsxTk40Ryt2YU1YM21QdEQyNkJ2VzExUXhPVG5ZdTVPSEM5ZWh6alhaQktkZVB5cHRzR1Z0LzJvbFpCUXloVXpJYStrUkdpa2NidWNRS3lzdU9DZ1UwemVoZXFoUmNUWXE5K01kQmg1Wk1xbXJDbUhGTjlDRzhUbVFwSlEvNHVqZExKNlFNUTFIR0RKaTE0M1BjYXRTdkk9IiwiaSI6InAwajIwM0FTZmV4RnljWlkifQ==) :lock:
* [Responsive mockups for screens delivered in Sprint 1](https://jstrieb.github.io/link-lock/#eyJ2IjoiMC4wLjEiLCJlIjoiRHo3c3pkcXRRTkdyRXdNUFdpU0pDeCtiQ2U1YlJEdGU3ellTaHVmKzE4ckhKWDhTMXpjT1h1MHlKd2NHMXlyT0M3Z0RGQ0U2Wk1kNytKUlZtUE8vNVVUaXU4OGFEUy9ybEo1TVBtV042V2haNUgxVU5aTVY3K1VYK3RlWXFPek56bzBsSTNLQnRBQ3dPajFYeW5BV3BZUUx5Zz09IiwiaSI6IkpiSUJzV3hiRGRXeWIxamUifQ==) :lock: 

### Sprint 1 Remediation:

- Updates in respect to the review conducted in the Sprint 1 PR. This will be part of the Sprint 2 review due to overlap in the remediation work of the [Sprint 1 PR](https://github.com/HHS/TANF-app/pull/41).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,47 +1,163 @@
...
+      - test-backend
+      - test-ui
+      - deploy:
+          requires:
+            - test-backend
+            - test-ui
+          filters:
+            branches:
+              only:
+                - sprint1-pr-review
```
Comment (lauraGgit) [inline .circleci/config.yml:163]: @RafterGit seems like this should be changed back to `raft-tdp-main`

File: delete-cloudgov-docker.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+YESIREALLYWANTTODOTHIS=${3}
+
+if [ -z "$YESIREALLYWANTTODOTHIS" ] ; then
+	echo "DANGER!  Not deleting unless you set the magic variable.  This deletes everything in your current space."
+	exit 1
+fi
+
+cf delete $CGHOSTNAME_BACKEND -r -f
+cf delete $CGHOSTNAME_FRONTEND -r -f
+cf delete-service db-raft -f
```
Comment (lauraGgit) [inline delete-cloudgov-docker.sh:19]: This needs to be updated to `tdp-db`

File: tdrs-frontend/.snyk [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  SNYK-JS-NODESASS-540980:
+    - node-sass:
+        reason: This is only used in compilation and wont be part for deployment.
+        expires: '2020-09-27T00:14:37.144Z'
+  SNYK-JS-NODESASS-540998:
+    - node-sass:
+        reason: This is only used in compilation and wont be part for deployment.
+        expires: '2020-09-27T00:14:37.145Z'
+  SNYK-JS-NODEFORGE-598677:
+    - react-scripts > webpack-dev-server > selfsigned > node-forge:
```
Comment (lauraGgit) [inline tdrs-frontend/.snyk:70]: selfsigned probably isn't used since we're serving on cloud.gov. you might want to update the policy to note that.

**Comment by lauraGgit** [comment]
@iamjolly @lfrohlich I've reviewed the Sprint 2 code and it is acceptable from my perspective. 

- [X] Government Tech review @lauraGgit 
- [  ] 18F COR accept @iamjolly 
- [  ] HHS ACF Product Owner accept & merge @lfrohlich 

# Government Acceptance

# QASP Deliverables 
## Deliverable 1: Accepted Features
* As a user I need to log in (frontend): ACs validated
* As a user, I can log out from the UI (mostly frontend): AC validated
* As a user I need to log in (backend): AC validated
* As a developer I need scaffolding for the backend so I can contribute to the project: AC validated
* UI Scaffolding Updates based on PR feedback: AC validated
* Set up CI/CD pipeline for OWASP Zap, Pa11y, and Code Coverage Reporting: AC validated
 
# Deliverable 2-7: 
Meets QASP requirements - [Sprint 2 updates](https://github.com/raft-tech/TANF-app/pull/198) address comments outlined in [Review comments](https://github.com/HHS/TANF-app/pull/47#pullrequestreview-479829617)

# Deliverable 8:
 No user research deliverables were completed this sprint, though comments provided on in progress work.

**Comment by iamjolly** [comment]
> @iamjolly @lfrohlich I've reviewed the Sprint 2 code and it is acceptable from my perspective.

@lauraGgit - Is there a place to see the working code in a dev/staging environment?

**Comment by lauraGgit** [comment]
The `ci/circleci: build Expected — Waiting for status to be reported` can be ignored because that is an artifact of a now out of date circleCI configuration.

**Comment by iamjolly** [comment]
I ran this locally and feel like it's good for now for COR approval with the following notes to address in the next sprint(s):

- I can see that the pa11y test passes in circleci, but I am finding accessibility issues on the front-end interface that should be caught if pa11y was running correctly.
  - TODO: Review pa11y configuration and ensure the tests are running and returning accurate results. This can be in the next sprint, and should continue in all following sprints.
- I do not know if a manual accessibility review was performed on the code for this sprint. This is essential to follow automated testing, as [automated accessibility scans only catch a small subset of possible accessibility issues](https://www.smashingmagazine.com/2018/09/importance-manual-accessibility-testing/). 
  - TODO: Review all code rendered in the browser for accessibility via automated tools and a manual process such as DHS Trusted Tester or Accessibility Insights for Web.

**Comment by lfrohlich** [comment]
re: @iamjolly 's comments plan to [define manual testing practices](https://app.zenhub.com/workspaces/tdrs-product-backlog-5f2c6cdc7c0bb1001bdc43a5/issues/raft-tech/tanf-app/173) in this next sprint and will also check on automated accessibility testing. 
In order to allow sufficient time for government review, PRs should be submitted the Friday before the end of the sprint.

**Pull request merged**